### PR TITLE
Switch to using location argument

### LINKED ISSUE #9: Use google_container_cluster location argument
There is now a `location` argument, which can be either a region or zone. https://www.terraform.io/docs/providers/google/r/container_cluster.html#location

This means we can remove the workaround we have using both a regional and zonal definition using `count` to control which is created.

### PR DESCRIPTION
**Pull request by wwwil**
Switch to using location argument, remove separate regional and zonal definitions, update everything accordingly.

Fixes #9 

Should merge #11 and #10 first and rebase

/cc munnerz

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,8 +12,9 @@ These are:
 
 - GCP project ID
 - Cluster name
-- GCP region
-- GCP zone (for a zonal cluster)
+- GCP location
+  - Specify a zone for a zonal cluster
```
Comment (munnerz) [inline README.md]: Maybe put `e.g. europe-west1-a` on this line, and `europe-west1` on the next?

File: example/variables.tf [hunk 2]
```diff
@@ -22,18 +22,14 @@ variable "cluster_name" {
...
-# each of three zones of the region.
-variable "gcp_region" {
-  type    = "string"
-  default = ""
+# The location (region or zone) in which the cluster master will be created,
+# as well as the default node location. If you specify a zone (such as
+# us-central1-a), the cluster will be a zonal cluster with a single cluster
+# master. If you specify a region (such as us-west1), the cluster will be a
+# regional cluster with multiple masters spread across zones in the region,
+# and with default node locations in those zones as well.
```
Comment (munnerz) [inline example/variables.tf]: `... and with default node locations in those zones as well.`

I'm not too sure what this means, as an outsider 😄

**Comment by wwwil** [comment]
I think this is now good to merge, I've also addressed the feedback!

/cc munnerz

**Comment by munnerz** [comment]
Much clearer 😄 

/lgtm

**Pull request merged**